File: frontend/src/types/evolution.ts
// 进化配置
export interface EvolutionConfig {
  population_size: number;
  max_generations: number;
  mutation_rate: number;
  crossover_rate: number;
  elite_rate: number;
  selection_method: 'tournament' | 'roulette' | 'rank';
  tournament_size?: number;
  fitness_threshold?: number;
  stagnation_limit?: number;
  diversity_threshold?: number;
}

// 进化状态
export type EvolutionStatus = 'idle' | 'running' | 'paused' | 'stopped' | 'completed' | 'error';

// 数字细胞
export interface DigitalCell {
  id: string;
  generation_id: string;
  gene_sequence: string;
  generated_code: string;
  fitness_score: number;
  parent_id?: string;
  mutation_rate: number;
  created_at: string;
  updated_at: string;
}

// 代数信息
export interface Generation {
  id: string;
  generation_number: number;
  population_size: number;
  average_fitness: number;
  best_fitness: number;
  worst_fitness: number;
  completed_at?: string;
  created_at: string;
  updated_at: string;
  cells?: DigitalCell[];
}

// 进化统计
export interface EvolutionStats {
  current_generation: number;
  total_generations: number;
  total_cells: number;
  best_fitness_ever: number;
  average_fitness_trend: number[];
  best_fitness_trend: number[];
  diversity_trend: number[];
  stagnation_count: number;
  runtime_seconds: number;
  cells_per_second: number;
  api_calls_total: number;
  api_cost_total: number;
}

// 进化控制请求
export interface EvolutionControlRequest {
  action: 'start' | 'stop' | 'pause' | 'resume' | 'reset';
  config?: EvolutionConfig;
}

// 意识状态
export interface ConsciousnessState {
  is_active: boolean;
  current_focus: string;
  reflection_depth: number;
  awareness_level: number;
  last_reflection: string;
  insights: string[];
  concerns: string[];
  suggestions: string[];
  emotional_state: 'curious' | 'focused' | 'concerned' | 'excited' | 'contemplative';
  updated_at: string;
}

// WebSocket消息类型
export interface WebSocketMessage {
  type: string;
  data: any;
  timestamp: string;
}

// 进化事件类型
export type EvolutionEventType = 
  | 'evolution_started'
  | 'evolution_stopped'
  | 'evolution_paused'
  | 'evolution_resumed'
  | 'evolution_reset'
  | 'generation_completed'
  | 'cell_evaluated'
  | 'fitness_improved'
  | 'consciousness_updated'
  | 'error_occurred';

// 进化事件数据
export interface EvolutionEvent {
  type: EvolutionEventType;
  data: {
    generation?: Generation;
    cell?: DigitalCell;
    stats?: EvolutionStats;
    consciousness?: ConsciousnessState;
    error?: string;
    message?: string;
  };
  timestamp: string;
}

// API响应类型
export interface ApiResponse<T = any> {
  success: boolean;
  data?: T;
  error?: string;
  message?: string;
}

// 分页响应
export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

// 错误类型
export interface ApiError {
  code: string;
  message: string;
  details?: any;
}

// 性能指标
export interface PerformanceMetrics {
  cpu_usage: number;
  memory_usage: number;
  disk_usage: number;
  network_io: number;
  active_connections: number;
  request_rate: number;
  error_rate: number;
  response_time: number;
}

// 系统健康状态
export interface SystemHealth {
  status: 'healthy' | 'warning' | 'critical';
  services: {
    database: 'up' | 'down' | 'degraded';
    redis: 'up' | 'down' | 'degraded';
    websocket: 'up' | 'down' | 'degraded';
    oracle: 'up' | 'down' | 'degraded';
    executor: 'up' | 'down' | 'degraded';
  };
  metrics: PerformanceMetrics;
  last_check: string;
}

File: backend/app/schemas/evolution.py
#!/usr/bin/env python3
"""
进化相关的Pydantic schemas
"""
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, Field

class EvolutionConfig(BaseModel):
    """进化配置"""
    population_size: int = Field(default=50, ge=10, le=1000, description="种群大小")
    mutation_rate: float = Field(default=0.1, ge=0.0, le=1.0, description="变异率")
    crossover_rate: float = Field(default=0.8, ge=0.0, le=1.0, description="交叉率")
    elite_size: int = Field(default=5, ge=1, le=50, description="精英个体数量")
    max_generations: Optional[int] = Field(default=None, ge=1, description="最大代数")

class EvolutionStatus(BaseModel):
    """进化状态"""
    is_running: bool = Field(description="是否正在运行")
    current_generation: int = Field(description="当前代数")
    population_size: int = Field(description="种群大小")
    best_fitness: Optional[float] = Field(description="最佳适应度")
    avg_fitness: Optional[float] = Field(description="平均适应度")
    total_evaluations: int = Field(description="总评估次数")
    start_time: Optional[datetime] = Field(description="开始时间")
    elapsed_time: Optional[float] = Field(description="已用时间（秒）")

class DigitalCellResponse(BaseModel):
    """数字细胞响应"""
    id: str = Field(description="细胞ID")
    generation_id: int = Field(description="所属代数ID")
    genome: str = Field(description="基因序列")
    generated_code: Optional[str] = Field(description="生成的代码")
    fitness_score: Optional[float] = Field(description="适应度分数")
    parent1_id: Optional[str] = Field(description="父代1 ID")
    parent2_id: Optional[str] = Field(description="父代2 ID")
    mutation_rate: Optional[float] = Field(description="变异率")
    created_at: datetime = Field(description="创建时间")
    
    class Config:
        from_attributes = True

class GenerationResponse(BaseModel):
    """代数响应"""
    id: int = Field(description="代数ID")
    generation_number: int = Field(description="代数编号")
    population_size: int = Field(description="种群大小")
    avg_fitness: Optional[float] = Field(description="平均适应度")
    max_fitness: Optional[float] = Field(description="最高适应度")
    min_fitness: Optional[float] = Field(description="最低适应度")
    created_at: datetime = Field(description="创建时间")
    completed_at: Optional[datetime] = Field(description="完成时间")
    digital_cells: Optional[List[DigitalCellResponse]] = Field(description="数字细胞列表")
    
    class Config:
        from_attributes = True

class EvolutionControlRequest(BaseModel):
    """进化控制请求"""
    action: str = Field(description="操作类型: start, pause, resume, stop, reset")
    config: Optional[EvolutionConfig] = Field(description="进化配置")

class EvolutionStatsResponse(BaseModel):
    """进化统计响应"""
    total_generations: int = Field(description="总代数")
    total_cells: int = Field(description="总细胞数")
    best_fitness_ever: Optional[float] = Field(description="历史最佳适应度")
    avg_fitness_trend: List[float] = Field(description="平均适应度趋势")
    max_fitness_trend: List[float] = Field(description="最高适应度趋势")
    api_cost_total: float = Field(description="总API成本")
    execution_time_total: float = Field(description="总执行时间")